compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -856,11 +864,63 @@ impl<'hir> LoweringContext<'_, 'hir> {
             // Lower outside new scope to preserve `is_in_loop_condition`.
             let fn_decl = this.lower_fn_decl(decl, None, FnDeclKind::Closure, None);
 
+            if let &ClosureBinder::For { span: for_sp, .. } = binder {
```

Comment:
I'm not sure this is the best place to make this test.  Should it go in `rustc_resolve::late::lifetimes`?

Reply:
Moved it in b8b1efda1384634e7a31b19567245abaee2af046